service/controller/resource/ipam/resource.go

```diff
@@ -24,6 +25,7 @@ const (
 type Config struct {
 	Checker   Checker
 	Collector Collector
+	K8sClient k8sclient.Interface
```

Comment:
The `ipam` handler needs the k8s client now in order to lookup the custom CIDR. A better design would have cost us unproportionally more time to refactor the whole shebang.